core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -330,7 +343,7 @@ object UnitActions {
         val globalPillageYield = Stats()
         val toCityPillageYield = Stats()
         val closestCity = unit.civInfo.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
-        val improvement = tile.ruleset.tileImprovements[tile.improvement]!!
+        val improvement = if (tile.improvement == null) tile.ruleset.tileImprovements[tile.roadStatus.name]!! else tile.ruleset.tileImprovements[tile.improvement]!!
```

Comment:
See this? THIS line should be a function in TileInfo, and then we can get rid of ALL the if/elses everywhere else!
Just run 'getImprovementToPillage()', and it will either return
- the improvement,
- the road improvement, or
- null

Reply:
Added function. Will need to review for cleanups

Reply:
Cleanups done